dashboard/utils/fine_tuning/rendering.ts

```diff
@@ -0,0 +1,32 @@
+import { InputMessageContent, Role } from "utils/clickhouse";
+import { JsExposedEnv } from "../minijinja/pkg";
+
+/**
+ * Renders a message using the provided environment and content.
+ *
+ * @param env - The Minijinja environment for rendering (should contain templates for "system", "user" and "assistant" if and only
+ *                                                       if the function takes structured input)
+ * @param role - The role of the message sender (must be "user" or "assistant")
+ * @param content - Array of message content blocks (must contain exactly one text block)
+ * @returns The rendered message string
+ * @throws {Error} If role is invalid, content length is not 1, or content type is not text
+ */
+export function render_message(
+  env: JsExposedEnv,
+  role: Role,
+  content: InputMessageContent[],
+) {
+  if (role !== "user" && role !== "assistant") {
+    throw new Error('Role must be either "user" or "assistant"');
+  }
```

Comment:
Not needed since role is an enum?

Reply:
still getting used to TypeScript semantics

Reply:
removed